src/main/java/de/tum/in/www1/artemis/service/ResultService.java

```diff
@@ -107,6 +110,26 @@ public void onResultNotifiedNew(Participation participation, Object requestBody)
         log.info("Received new build result (NEW) for participation " + participation.getId());
 
         Result result = continuousIntegrationService.get().onBuildCompletedNew(participation, requestBody);
+
+        if (participation.getExercise() instanceof ProgrammingExercise) {
```

Comment:
extract this into its own method, e.g. extractTestCases

Reply:
You are right!